src/course-search/FacetDropdown.jsx

```diff
@@ -1,33 +1,79 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import { Dropdown } from '@edx/paragon';
+import { Dropdown, Input } from '@edx/paragon';
 import classNames from 'classnames';
 
-const FacetDropdown = ({ title, items, isBold }) => (
-  <div className="facet-list">
-    <Dropdown className="mb-0 mr-md-3">
-      <Dropdown.Toggle
-        className={
-          classNames(
-            'bg-white', 'text-capitalize', 'rounded-0', 'border-0',
-            'd-flex', 'justify-content-between', 'align-items-center', 'text-dark',
-            { 'font-weight-bold': isBold },
-          )
-        }
-      >
-        {title}
-      </Dropdown.Toggle>
-      <Dropdown.Menu>
-        {items}
-      </Dropdown.Menu>
-    </Dropdown>
-  </div>
-);
+const FacetDropdown = ({
+  title,
+  items,
+  isBold,
+  typeahead,
+  searchForItems,
+}) => {
+  const handleSearch = (value) => {
+    // when user is erasing the input and input is empty we need to reset the filtering
+    if (value.length >= typeahead.minLength || value.length === 0) {
+      searchForItems(value);
```

Comment:
Where are we getting this function from? Is it Algolia? Just to confirm, we want it on all of our facet dropdowns, not just the new one?

Reply:
Oh, I see, you're only setting `typeahead` for the one facet. 
Is there a strong reason *not* to add this for our other dropdowns? 😄

Reply:
yes, we get `searchForItems` from Algolia. Please see https://www.algolia.com/doc/api-reference/widgets/refinement-list/react/#connector-param-provided-searchforitems

I am not sure if we want to add the typeahead for all the dropdowns.  😄

Reply:
I suspect we do; I'll ask UX next time I meet with them. Don't worry about it for this PR.